janusgraph-core/src/main/java/org/janusgraph/core/VertexConnection.java

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.core;
+
+/**
+ * @author Jan Jansen (jan.jansen@gdata.de)
+ */
+public class VertexConnection {
```

Comment:
Need a clear description on this class since it is new terminology. In the `Schema Constraints` docs, it refers to this as `Connection`. Perhaps this should be named `Connection` instead of `VertexConnection`. Where is the outgoing vertex label?